src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,210 @@
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(true);
+    }
+
+    @Test
+    void whenDataIsNullAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(null)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenNewStructureDoesNotExistAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(ImmutableMap.of("data", "someData"))
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenOldApplicantStructureIsMigratedShouldReturnNewApplicantStructure() {
+        Map<String, Object> data = new HashMap<>();
+
+        data.put("applicant", createOldApplicant());
+
+        CaseData migratedData = service.migrate(data);
+
+        ApplicantParty actualValue = migratedData.getApplicants().get(0).getValue().getParty();
+        ApplicantParty expectedValue = createNewApplicant().getParty();
+
+        assertThat(migratedData.getApplicant()).isNull();
+
+        assertThat(actualValue.getPartyId()).isNotNull();
+        assertThat(actualValue.getPartyType()).isNull();
+        assertThat(actualValue.getOrganisationName()).isEqualTo(expectedValue.getOrganisationName());
+        assertThat(actualValue.getPbaNumber()).isEqualTo(expectedValue.getPbaNumber());
+        assertThat(actualValue.getAddress()).isEqualTo(expectedValue.getAddress());
+        assertThat(actualValue.getTelephoneNumber()).isEqualTo(expectedValue.getTelephoneNumber());
+        assertThat(actualValue.getJobTitle()).isEqualTo(expectedValue.getJobTitle());
+        assertThat(actualValue.getMobileNumber()).isEqualTo(expectedValue.getMobileNumber());
+        assertThat(actualValue.getEmail()).isEqualTo(expectedValue.getEmail());
+    }
+
+    @Test
+    void whenOldApplicantContainsNullValuesShouldReturnNewApplicantWithPartyIdAndNullValues() {
+        Map<String, Object> data = new HashMap<>();
+
+        OldApplicant applicant = OldApplicant.builder()
+            .name(null)
+            .email(null)
+            .mobile(null)
+            .address(null)
+            .jobTitle(null)
+            .telephone(null)
+            .personToContact(null)
+            .pbaNumber(null)
+            .build();
+
+        data.put("applicant", applicant);
+
+        CaseData migratedData = service.migrate(data);
+
+        ApplicantParty actualValue = migratedData.getApplicants().get(0).getValue().getParty();
+
+        assertThat(migratedData.getApplicant()).isNull();
+        assertThat(actualValue.getPartyId()).isNotNull();
+        assertThat(actualValue.getPartyType()).isNull();
```

Comment:
In that case party type should not be null either.

REQ

Reply:
added assertions around partyType = ORGANISATION